serokell-style.md

```diff
@@ -288,372 +300,261 @@ Prefer two-line imports for such standard containers.
 infixl 5 />
 (/>) :: Uri -> PathPiece -> Uri
 ```
 
 ### Pragmas
 
-Put pragmas immediately following the function they apply to.
-Example:
+If you need to use pragmas, you _must_ put them immediately
+after the function that they apply to. Example:
 
 ```haskell
 id :: a -> a
 id x = x
 {-# INLINE id #-}
 ```
 
-In case of data type definitions you must put the pragma before
+For data type definitions, you _must_ put the pragma before
 the type it applies to. Example:
 
 ```haskell
 data Array e = Array
-    {-# UNPACK #-} !Int
-    !ByteArray
+  {-# UNPACK #-} Int
+  ByteArray
 ```
 
 ### List Declarations
 
-Align the elements in the list.  Example:
+Depending on the length of list elements, you _should_ either keep the list on one
+line or put each element on a separate line. In the latter case, put a trailing `]`
+on a separate line, like `}` when you format records. Example:
 
 ```haskell
-exceptions =
-    [ InvalidStatusCode
-    , MissingContentHeader
-    , InternalServerError
-    ]
-```
+numbers = [1, 2, 4]
 
-Optionally, you can skip the first newline. Use your judgement.
-
-```haskell
-directions = [ North
-             , East
-             , South
-             , West
-             ]
+exceptions =
+  [ InvalidStatusCode
+  , MissingContentHeader
+  , InternalServerError
+  ]
 ```
 
 ### Hanging Lambdas
 
-Don't insert a space after a lambda.
+You _must not_ insert a space after a lambda.
 
 You may or may not indent the code following a "hanging" lambda.  Use
 your judgement. Some examples:
 
 ```haskell
 bar :: IO ()
-bar = forM_ [1, 2, 3] $ \n -> do
-          putStrLn "Here comes a number!"
-          print n
+bar =
+  forM_ [1, 2, 3] $ \n -> do
+    putStrLn "Here comes a number!"
+    print n
 
 foo :: IO ()
-foo = alloca 10 $ \a ->
-      alloca 20 $ \b ->
-      cFunction a b
+foo =
+  alloca 10 $ \a ->
+  alloca 20 $ \b ->
+  cFunction a b
 ```
 
 ### If-then-else clauses
 
-Generally, guards and pattern matches should be preferred over _if-then-else_
+Generally, guards and pattern matches should be preferred over `if-then-else`
 clauses, where possible.  Short cases should usually be put on a single line
 (when line length allows it).
 
 When writing non-monadic code (i.e. when not using `do`) where guards
-and pattern matches can't be used, you can align _if-then-else_ clauses
+and pattern matches cannot be used, you _may_ align `if-then-else` clauses
 like you would normal expressions:
 
 ```haskell
-foo = if ...
-      then ...
-      else ...
+foo =
+  if ...
+  then ...
+  else ...
 ```
 
-Or you can align _if-then-else_ in different style inside lambdas.
+You _may_ align `if-then-else` in different style inside lambdas.
 
 ```haskell
-foo = bar $ \qux -> if predicate qux
+foo =
+  bar $ \qux -> if predicate qux
     then doSomethingSilly
     else someOtherCode
 ```
 
-You can also write _if-then-else_ in imperative style inside do blocks
+You _may_ also write `if-then-else` in imperative style inside do blocks.
```

Comment:
Actually, your way seems better than what is currently written in the guide, because `condition` may become long and ultimately split over multiple lines. I will push an update with your example and see if anyone has objections to it.

Reply:
No objection, this style is fine.